### PR TITLE
Siglip: add `_no_split_module`

### LINKED ISSUE #31400: SiqlipVisionModel does not support "device map= auto": no split modules`attribute
### Feature request

How to make SiglipVisionModel can support Auto map to map to multiple GPUs.

### Motivation

Currently, using MLLM with siglip, the whole model might need automap, since the vision encoder part is SiglipVisionModel, if it doesn;t support , would be a big problme

Also, would consider add flashattention support for Siglip?

### Your contribution

None for now

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

Fixes #31400. Adds a '_no_split_module' in SigLipModel by combining those specified in Text and Vision components. 

Otherwise, when we call SiglipModel with `device_map` an error will be thrown because Text/Vision components are not children of SiglipModel and we need to add `_no_split_module`

### CODE REVIEW

File: src/transformers/models/siglip/modeling_siglip.py [hunk 1]
```diff
@@ -1018,6 +1018,13 @@ def forward(
 @add_start_docstrings(SIGLIP_START_DOCSTRING)
 class SiglipModel(SiglipPreTrainedModel):
     config_class = SiglipConfig
+    _no_split_modules = [
+        "SiglipTextEmbeddings",
+        "SiglipEncoderLayer",
+        "SiglipVisionEmbeddings",
+        "SiglipEncoderLayer",
+        "SiglipMultiheadAttentionPoolingHead",
+    ]
```
Comment (amyeroberts) [inline src/transformers/models/siglip/modeling_siglip.py:1027]: What happens if we have modules defined in `_no_split_modules`  which aren't in the model? It probably makes more sense to move this into the `PretrainedModel` definition than repeat e.g. `"SiglipVisionEmbeddings"` for all models which use the vision components

**Review by amyeroberts — COMMENTED** [review]
Thanks for fixing! 

General comment / question about the best approach

File: src/transformers/models/siglip/modeling_siglip.py (hunk 1 above)
Comment (zucchini-nlp) [reply src/transformers/models/siglip/modeling_siglip.py]: Great idea! Didn't think that extra `split_modules` will not affect anything. Moved to SiglipPretrained model and ran tests. Had to skip offloading, same way it's skipped in SigLipVision

**Review by amyeroberts — APPROVED** [review]
Thanks for adding!

**Pull request merged**